crates/ruff/src/rules/flake8_copyright/rules/copyright_header_absent.rs

```diff
@@ -0,0 +1,47 @@
+use ruff_diagnostics::Violation;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::newlines::Line;
+
+use crate::settings::Settings;
+
+use lazy_regex::Regex;
+#[violation]
+pub struct HeaderLacksCopyright;
+
+impl Violation for HeaderLacksCopyright {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Copyright notice not present")
+    }
+}
+/// ## What it does
+/// Checks for Copyright Header to exist within at the top of a file within `copyright_min_file_size chars`
+/// format Copyright (C) <year> <author>
+///
+/// Error code C801
+pub(crate) fn copyright_header_absent(
```

Comment:
Is it necessary that this rule runs on every line? Could we instead call the rule just once, passing the full source code? Should this rule only run on comments (or doc-strings)? rather than on all source code?

Reply:
![image](https://github.com/charliermarsh/ruff/assets/31294356/f2e95a9d-e4ea-47b5-a580-2f4104645a46)
I initially asked on discord. I think it would more efficient as well to pass the full source code as well.

Is there any rule that currently does this pattern or should I just create a separate portion in check_physical_lines to pass the entire source code?

Reply:
@charliermarsh what's your take?

Reply:
Yeah I'd suggest rewriting to call once per file, especially since it's limited to the first N characters anyway.